src/test/java/bookbob/BookBobTest.java

```diff
@@ -450,34 +459,39 @@ void testTextConverterFullInformation() {
         diagnosis.add("Tummy bug");
         ArrayList<String> medications = new ArrayList<>();
         medications.add("Gaviscon");
-        String dateTimeString = "2024-10-21 15:48";
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        String dateTimeString = "21-10-2024 15:48";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
         LocalDateTime visitDateTime = LocalDateTime.parse(dateTimeString, formatter);
         ArrayList<Visit> visits = new ArrayList<>();
         visits.add(new Visit(visitDateTime, diagnosis, medications));
+        ArrayList<String> allergies = new ArrayList<>();
+        allergies.add("Peanuts");
+        ArrayList<String> medicalHistory = new ArrayList<>();
+        medicalHistory.add("History of gastritis");
+
         Patient patient = new Patient("John", "S9765432T", "87658976", "06071997",
-                "Bukit Gombak", "Peanuts", "Male", "History of gastritis", visits);
+                "Bukit Gombak", allergies, "Male", medicalHistory, visits);
         String output = fileHandler.convertPatientToOutputText(patient);
         assertEquals(output, "Name: John | NRIC: S9765432T | Phone Number: 87658976 | " +
-                "Date_Of_Birth: 06071997 | Home Address: Bukit Gombak | Allergy: Peanuts " +
-                "| Sex: Male | Medical History: History of gastritis | Visit: " +
-                "[2024-10-21 15:48, Diagnosis: [Tummy bug], " +
+                "Date_Of_Birth: 06071997 | Home Address: Bukit Gombak | Allergy: [Peanuts] " +
+                "| Sex: Male | Medical History: [History of gastritis] | Visit: " +
+                "[21-10-2024 15:48, Diagnosis: [Tummy bug], " +
                 "Medications: [Gaviscon]];", output);
     }
 
     //@@author PrinceCatt and coraleaf0602
     @Test
     void testTextConverterPartialInformation() {
-        String dateTimeString = "2024-10-21 15:48";
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        String dateTimeString = "21-10-2024 15:48";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
         LocalDateTime visitDateTime = LocalDateTime.parse(dateTimeString, formatter);
-        List<Visit> visits = new ArrayList<>();
+        ArrayList<Visit> visits = new ArrayList<>();
```

Comment:
Good that you recognised that List<Visit> cannot be assigned to ArrayList<Visit>